galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyRoute.java

```diff
@@ -0,0 +1,196 @@
+        try{
+           Map<String, String> properties = getAllProperties(namespace);
+           for (Map.Entry<String, String> entry : properties.entrySet()) {
+               String key = entry.getKey().toString();
+               if (key.equals(propertyName)){
+                   return entry;
+               }
+           }
+        }catch (Exception e){
+            ServletError error = new ServletError(GAL5017_INVALID_NAMESPACE_ERROR,namespace);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+        }
+        return null;
+    }
+
+    private boolean checkPropertyExists (String namespace, String propertyName) throws FrameworkException{
+        Map.Entry<String, String> entry = retrieveProperty(namespace, propertyName);
+        if( entry  == null){
+            return false;
+        }
+        return true;
+    }
+
+    private void checkRequestHasContent(HttpServletRequest request, String pathInfo) throws InternalServletException{
+        boolean valid = false;
+        try{
+            if (request.getContentLength() >0){
+                valid = true;
+            }
+        }catch (NullPointerException e ){
+            //Catch the NullPointerException (empty request body) 
+            //Exception is thrown by the if below
+        }  
+        if (!valid){
+            ServletError error = new ServletError(GAL5411_NO_REQUEST_BODY,pathInfo);  
+            throw new InternalServletException(error, HttpServletResponse.SC_LENGTH_REQUIRED);
+        }
+    }
+
+    /*
+     * Handle Get Request
+     */
+    @Override
+    public HttpServletResponse handleRequest(String pathInfo, QueryParameters queryParams,HttpServletRequest req, HttpServletResponse response) throws ServletException, IOException, FrameworkException {
+        String namespace = getNamespaceFromURL(pathInfo);
+        String propertyName = getPropertyNameFromURL(pathInfo);
+        String  property= getProperty(namespace,propertyName);
+		return getResponseBuilder().buildResponse(response, "application/json", property, HttpServletResponse.SC_OK); 
+    }
+
+    private String getProperty(String namespace, String propertyName) throws FrameworkException {
+        Map.Entry<String, String> entry = retrieveProperty(namespace, propertyName);
+        JsonArray propertyArray = new JsonArray();
+        if (entry != null){
+            JsonObject cpsProp = new JsonObject();
+            cpsProp.addProperty("name", entry.getKey());
+            cpsProp.addProperty("value", getProtectedValue(entry.getValue(),namespace));
+            propertyArray.add(cpsProp);
+        }
+        return gson.toJson(propertyArray);
+    }
+
+    /*
+     * Handle Put Request
+     */
+    @Override
+    public HttpServletResponse handlePutRequest(String pathInfo, QueryParameters queryParameters, HttpServletRequest request , HttpServletResponse response)
```

Comment:
Put requests should fail if the property isn't there already.
Post requests should fail if the property is already there.